### PR TITLE
Test controllers

### PR DESCRIPTION
**Pull request by dgsierra**
Add:

- Test for Post:

1.   - Status
2.   - response
3.   - template

- Test for User:

1.   - Status
2.   - response
3.   - template

Fix:

 - Install with Rspec, updated to a new version for rails 7

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
```
Comment (SajjadAhmad14) [inline spec/requests/users_spec.rb:2]: User Spec file should include tests for both `show` and `index` that ensures correct view template is rendered.
`Example : expect(response).to render_template(:show)`

User Spec file should include test for both `show` and `index` that ensures the response body includes the correct placeholder text.
`Example : expect(response.body).to include("Welcome to user index page")`

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe 'Post', type: :request do
+  describe 'Get index route of users controller' do
```
Comment (SajjadAhmad14) [inline spec/requests/posts_spec.rb:4]: Post Spec file should include tests for both `show` and `index` that ensures correct view template is rendered.
`Example : expect(response).to render_template(:show)`

Post Spec file should include test for both `show` and `index` that ensures the response body includes the correct placeholder text.
`Example : expect(response.body).to include("Welcome to user index page")`

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Daniel, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/requests/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        get users_path(1)
+      end
+
+      it 'Returns success status for show routes' do
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'Returns or renders the show template' do
+        expect(response).to render_template(:index)
+      end
```
Comment (BenMukebo) [inline spec/requests/posts_spec.rb:21]: - [ ] Great job so far 👏🏻 however as requested by the previous reviewer:
> Post Spec file should include tests for both `show` and `index` that ensures the correct view template is rendered.
> `Example : expect(response).to render_template(:show)`

- Actually, you're missing the test that ensures the correct view `show` template is rendered. Please add this ✍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @dgonzalesi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 3 above)
Comment (AnselemOdims) [reply spec/requests/posts_spec.rb:21]: - [ ] As requested by the previous reviewers, please include a test to ensure that the correct template `show` is being rendered.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @dgonzalesi , 
Good job implementing some of the changes requested by the previous reviewer.

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**